app/services/exchange_rates/create_old_files_service.rb

```diff
@@ -0,0 +1,188 @@
+# DELETE once this has been run
+module ExchangeRates
+  class CreateOldFilesService
+    MONTH_INDEX = 0
+    YEAR_INDEX = 1
+
+    def self.call
+      new.call
+    end
+
+    def call
+      old_file_dates.each do |date_hash|
+        month_name = date_hash[:edition_date].split(' ')[MONTH_INDEX]
+        month = Date::MONTHNAMES.index(month_name)
+        year = date_hash[:edition_date].split(' ')[YEAR_INDEX]
+
+        data_result = ::ExchangeRateCurrencyRate.for_month(month, year)
+
+        created_date = date_hash[:updated_date]
+
+        upload_data(data_result,
+                    :csv,
+                    ExchangeRates::CreateCsvService,
+                    created_date,
+                    month,
+                    year)
+        upload_data(data_result,
+                    :xml,
+                    ExchangeRates::CreateXmlService,
+                    created_date,
+                    month,
+                    year)
+      end
+    end
+
+  private
+
+    def upload_data(data_result, format, service_class, created_date, month, year)
```

Comment:
a lot of this code is duplicated from app/services/exchange_rates/upload_monthly_file_service.rb, wouldn't it be better to remove some of this duplication?

Reply:
The reason i moved it here was mainly because i thought this file would probably be deleted once it had been run once as shouldnt need to be run again?